thewalrus/operations.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Operations
+==========
+
+**Module name:** :mod:`thewalrus.operations`
+
+.. currentmodule:: thewalrus.operations
+
+Functions to construct the Fock representation of a Gaussian operation
+represented as a Symplectic matrix and complex displacements.
+
+
+Contains some Gaussian operations and auxiliary functions.
+
+Auxiliary functions
+-------------------
+
+.. autosummary::
+    n_two_mode_squeezed_vac
+    choi_expand
+
+Operations
+---------------
+
+.. autosummary::
+    fock_tensor
+"""
+
+from itertools import product
+import numpy as np
+from thewalrus import hafnian_batched
+from thewalrus.symplectic import expand
+from thewalrus.quantum import Amat
+
+
+def n_two_mode_squeezed_vac(n, r=np.arcsinh(1.0)):
```

Comment:
Two questions:

1. Isn't S2gate already in thewalrus.symplectic?

2. This symplectic transformation can be applied to any state, so why does the name have `vac`?

Reply:
1. Yes. This is just meant to be a quick way of constructing the tensor product of `S2gate `s  between modes `i` and `i+n`. It is just a lot faster to write the matrix directly than to compose it, especially since they all have the same squeezing parameter. It might be worth to fold this functionality inside the `fock_tensor` since it is only used there
2. You are right, the `vac` could be removed.

Reply:
ahhh, maybe you could put `tensor_` in the name or something, makes it more clear 🙂